### DISCUSSION TITLE
Wait for all LSPs to be installed in headless mode

### DISCUSSION THREAD

**Discussion by pose** [mason-org/mason.nvim#923] [Q&A]
Is there a way in `--headless`  mode to wait for all the LSPs to be installed without having to manually specify each of them? I'm using:

```
nvim --headless +Mason +"MasonInstall typescript-language-server" +MasonLog +"w! mason-install.log" +qall
```

but that doesn't work if `typescript-language-server` installs faster than the rest. I tried waiting for `:Mason` alone, but it's not blocking.

Thanks!

**Comment by williamboman** [comment]
Hey! `:MasonInstall` should run in a blocking fashion if executed in a headless environment (i.e. the command only yields back once all packages has finished installation).
